### PR TITLE
Raise error if duplicate fields across spec files

### PR DESCRIPTION
**Pull request by nmuesch**
### What does this PR do?

Slightly cleaner dict updates and detect + raise if there are duplicate components across files. 

### Motivation

Cleanup + more error detection

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Files changes must correspond to the primary purpose of the PR as described in the title (small unrelated changes should have their own PR)
- [ ] Feature or bugfix must have tests
- [ ] Git history [must be clean](https://github.com/DataDog/apigentools/blob/master/CONTRIBUTING.md#commit-messages)

### CODE REVIEW

File: apigentools/utils.py [hunk 1]
```diff
@@ -221,17 +232,17 @@ def write_full_spec(config, spec_dir, version, full_spec_file):
             else:
                 full_spec["paths"].update(loaded.get("paths", {}))
```
Comment (bkabrda) [inline apigentools/utils.py]: `paths` should also be checked for duplicates

File: apigentools/utils.py [hunk 2]
```diff
@@ -221,17 +232,17 @@ def write_full_spec(config, spec_dir, version, full_spec_file):
...
-                full_spec["components"]["callbacks"].update(loaded.get("components", {}).get("callbacks", {})),
                 full_spec["security"].extend(loaded.get("security", {}))
 
+                for field in COMPONENT_FIELDS:
+                    # Validate there aren't duplicate fields across files
+                    # Note: This won't raise an error if there is a duplicate component in a single file
+                    # That would alredy be deduped by the safe_load above.
+                    for key in loaded.get('components', {}).get(field, {}).keys():
+                        if key in full_spec.get('components', {}).get(field, {}):
+                            raise ValueError("Duplicate field {} found in spec. Exiting".format(key))
```
Comment (bkabrda) [inline apigentools/utils.py]: It would be awesome if this printed which 2 yaml files contain this duplicity.

File: apigentools/utils.py [hunk 3]
```diff
@@ -221,17 +232,17 @@ def write_full_spec(config, spec_dir, version, full_spec_file):
             else:
                 full_spec["paths"].update(loaded.get("paths", {}))
                 full_spec["tags"].extend(loaded.get("tags", []))
```
Comment (bkabrda) [inline apigentools/utils.py]: We should also check for duplicate tags, even if this is not dict - having tags with same names would still be wrong.

File: apigentools/utils.py (hunk 2 above)
Comment (nmuesch) [reply apigentools/utils.py]: At this point we have access to the file where we found the second occurrence , but we aren't keeping track of where we first found each field. While I agree seeing both file names would be useful, I'm not sure its worth the logic of keeping track of each field we see along the way. What do you think?

**Comment by nmuesch** [comment]
@bkabrda Thanks for the review! I think I've addressed/commented on everything, can you take another pass? 

I'm sure there is a cleaner way to do this and update all elements `component` or otherwise at once, but its a bit trickier than I expected 🙂

File: apigentools/utils.py (hunk 2 above)
Comment (bkabrda) [reply apigentools/utils.py]: Yeah, fair. We can always do it later.

**Pull request merged**